src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/BuildPropertyPage/TrimModeValueProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+[ExportInterceptingPropertyValueProvider("TrimMode", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+internal class TrimModeValueProvider : InterceptingPropertyValueProviderBase
+{
+    public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
```

Comment:
Sorry, I'm unfamiliar with how this actually works. Does setting the property value mean writing into the project file? Or is it setting it somehow in the internal project system build?

Reply:
This file is going to change in a bit with the given discussion upstairs, but to answer your question: this will intercept before we save it into the project file. So, in this case, I had it so that setting `TrimMode` would also set another property (but this will no longer be the case).